middleware.ts

```diff
@@ -1,7 +1,73 @@
-import {NextRequest, NextResponse} from 'next/server';
+import { NextRequest, NextResponse } from 'next/server';
+import { JwtPayload } from 'jsonwebtoken';
+import { verifyJwtToken } from './lib/server/auth';
+import { getAuthTokenServer } from '@/utils/getAuthTokenServer';
+import createMiddleware from 'next-intl/middleware';
+import { routing } from './i18n/routing';
+
+const locales = ['en-US', 'pt-BR'];
+const defaultLocale = 'en-US';
+
+interface JWTAccessToken extends JwtPayload {
+  id: number,
+  email: string,
+  givenName: string,
+  surName: string,
+  role: string,
+  languageId: string,
+  jti: string,
+  expiresIn: number,
+}
 
 // TODO: These routes will need to be updated.
-const excludedPaths = ['/login', '/signup', '/email', '/favicon.ico', '/_next', '/api'];
+const excludedPaths = ['/email', '/favicon.ico', '/_next', '/api'];
```

Comment:
Might need to figure out how to smooth this out with potentially multiple language pages too